### PR TITLE
initial signature verification implementation

### PR DESCRIPTION
**Pull request by jbowes**

### CODE REVIEW

File: lib/manifoldco_signature/signature.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+require 'base64url'
+require 'rbnacl/libsodium'
+
+module ManifoldcoSignature
+
+  class Signature
+
+    def initialize(req)
+      sig = req.env['HTTP_X_SIGNATURE'].split
```
Comment (ianlivingstone) [inline lib/manifoldco_signature/signature.rb:11]: We should probably fall back to the manifold master key as a default? Or perhaps that will lead to confusion.

Comment (ianlivingstone) [reply lib/manifoldco_signature/signature.rb:11]: Wrong place!

Comment (ianlivingstone) [reply lib/manifoldco_signature/signature.rb:11]: As an aside though, show we check the length of the resulting `sig` to be `3`, or only rely on the base64url decode to throw?

File: lib/manifoldco_signature/verifier.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require 'base64url'
+require 'rbnacl/libsodium'
+
+module ManifoldcoSignature
+  MASTER_KEY = "PtISNzqQmQPBxNlUw3CdxsWczXbIwyExxlkRqZ7E690"
```
Comment (ianlivingstone) [inline lib/manifoldco_signature/verifier.rb]: 👍

File: lib/manifoldco_signature/verifier.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  class Verifier
+
+    def initialize(master_key=MASTER_KEY)
+      dec = Base64URL.decode(master_key)
+      @master_key = RbNaCl::VerifyKey.new dec
```
Comment (ianlivingstone) [inline lib/manifoldco_signature/verifier.rb:10]: will this throw if the master key ins't the right length etc?

Comment (jbowes) [reply lib/manifoldco_signature/verifier.rb:10]: yup

File: lib/manifoldco_signature/signature.rb (hunk 1 above)
Comment (jbowes) [reply lib/manifoldco_signature/signature.rb:11]: It's all pretty dumb right now. well, it's correct, but not informative.

if anything is wrong with the b64 of the sigs/keys, or headers are missing, or what have you, the code will throw, and be caught above and treated as a failed verification.

Given that from a provider's POV all they'll be able to do is misconfigure the key, and we will throw if that happens and its not a valid key, that's really enough.

Comment (ianlivingstone) [reply lib/manifoldco_signature/signature.rb:11]: Gotcha

**Pull request merged**